src/python/pants/bin/pants_runner.py

```diff
@@ -59,11 +59,11 @@ def _should_run_with_pantsd(self, global_bootstrap_options: OptionValueContainer
     def scrub_pythonpath() -> None:
         # Do not propagate any PYTHONPATH that happens to have been set in our environment
         # to our subprocesses.
-        # Note that don't warn (but still scrub) if RUNNING_PANTS_FROM_SOURCES is set. This allows
-        # scripts that run pants directly from sources, and therefore must set PYTHONPATH, to mute
+        # Note that don't warn if RUNNING_PANTS_FROM_SOURCES is set. This allows scripts that
+        # run pants directly from sources, and therefore must set PYTHONPATH, to mute
         # this warning.
         pythonpath = os.environ.pop("PYTHONPATH", None)
-        if pythonpath and not os.environ.pop("RUNNING_PANTS_FROM_SOURCES", None):
+        if pythonpath and not os.environ.get("RUNNING_PANTS_FROM_SOURCES", "0") == "1":
```

Comment:
I'm not sure what the consequences are of leaving this set, but I can't think of negative ones offhand.

Reply:
I looked around and didn't see any at all. 🙂

Reply:
I grepped the codebase and didn't see any